app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,27 +10,49 @@ def self.call(*args)
       end
 
       def call
-        return if commented_on_welcome_thread? || received_notification?
+        return unless user
+
+        send_welcome_notification
+        send_authentication_notification
+      end
+
+      def send_welcome_notification
+        return if received_notification?(welcome_broadcast) || commented_on_welcome_thread?
 
         Notification.send_welcome_notification(user.id, welcome_broadcast.id)
       end
 
-      def received_notification?
-        Notification.exists?(notifiable: welcome_broadcast, user: user)
+      def send_authentication_notification
+        return if received_notification?(authentication_broadcast) || authenticated_with_both_services? || user.created_at > 1.day.ago
```

Comment:
I think we'd want `authenticated_with_both_services?` to come first here, right?

Just thinking through what would happen when line 26 evaluates for a user that has already authenticated with both services immediately on sign up (but hasn't gotten the notifications for authentication just yet):
- first, we evaluate `received_notification?`, which defines `@authentication_broadcast`
- if someone has a github identity, we find the twitter auth broadcast, and if someone doesn't have a github identity, we find the github auth broadcast and assign it to `@authentication_broadcast`:
```
missing_identity = user.identities.exists?(provider: "github") ? "twitter_connect" : "github_connect"
@authentication_broadcast ||= Broadcast.find_by(title: "Welcome Notification: #{missing_identity}")
```

But what happens if someone has got both a github and a twitter identity? in that case, `@authentication_broadcast` would evaluate to the twitter broadcast, and we'd be checking if the user had received the notification for it, and if they hadn't, we'd be sending them a notification for something that didn't really apply to them. If we made `authenticated_with_both_services?` the first check, I think we could get around this issue.

Reply:
Oh, good catch. This wasn't immediately obvious to me.

Reply:
I swapped it